### PR TITLE
Fixed the error message for a combination with a min quantity for sale

### LINKED ISSUE #21873: BO - Order page - The minimum quantity of a combination is not correct
#### Describe the bug

In the BO > Orders > Add new order page > the minimum quantity of a combination is not correct.

#### Expected behavior
image

#### Steps to Reproduce

1. Go to BO > Products page > Create a new product with combinations
3. For each combination set the Min. quantity for sale > 1
image
3. Go to BO > Order page > Add new Order
4. choose a customer
5. search the last product created
6. Try to add it to the cart
7. an alert is displayed, but the minimum quantity is NOK.

image

#### Additional information

* PrestaShop version: 1.7.7.x
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by Progi1984**
# Don't merge before the release of 1770

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Fixed the error message for a combination with a min quantity for sale
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #21873
| How to test?  | Cf #21873

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21980)

### CODE REVIEW

File: tests/Integration/Behaviour/Features/Context/Domain/CartFeatureContext.php [hunk 1]
```diff
@@ -201,6 +214,44 @@ public function addCustomizedProductToCarts(int $quantity, $productReference, $r
...
+        $combinationId = $this->productFeatureContext->getCombinationWithName($productName, $combinationName)->id;
+        $cartId = (int) SharedStorage::getStorage()->get($cartReference);
+
+        $this->lastException = null;
+        try {
+            $this->getCommandBus()->handle(
+                new UpdateProductQuantityInCartCommand(
+                    (int) $cartId,
+                    (int) $productId,
+                    (int) $quantity,
```
Comment (PierreRambaud) [inline tests/Integration/Behaviour/Features/Context/Domain/CartFeatureContext.php]:
```suggestion
                    $quantity,
```

**Review by PierreRambaud — DISMISSED** [review]

File: src/Adapter/Cart/CommandHandler/UpdateProductQuantityInCartHandler.php [hunk 2]
```diff
@@ -129,7 +129,7 @@ private function updateProductQuantityInCart(Cart $cart, UpdateProductQuantityIn
         // It seems that $updateResult can be -1,
         // when adding product with less quantity than minimum required.
         if ($updateResult < 0) {
-            $minQuantity = $command->getCustomizationId() ?
+            $minQuantity = $command->getCombinationId() ?
```
Comment (jolelievre) [inline src/Adapter/Cart/CommandHandler/UpdateProductQuantityInCartHandler.php]:
```suggestion
            $minQuantity = $combinationIdValue ?
```

Haha such a shitty bug 😅. Using the already present variable will reduce potential bugs even more

**Review by PierreRambaud — DISMISSED** [review]

**Review by jolelievre — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @Progi1984,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1Ob0MJHvSMILTrsTjI1CFjXiEHfVJ1L3l/view?usp=sharing

Thanks!

**Comment by Progi1984** [comment]
# Don't merge before the release of 1770

**Review by PierreRambaud — DISMISSED** [review]
Don't merge before the release of 1770

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**